staging/src/k8s.io/apiserver/pkg/admission/metrics/testutil_test.go

```diff
@@ -42,20 +33,18 @@ func expectFindMetric(t *testing.T, name string, expectedLabels map[string]strin
 	for _, mf := range metrics {
 		if mf.GetName() == name {
 			for _, metric := range mf.GetMetric() {
-				if labelsMatch(metric, expectedLabels) {
+				if testutil.LabelsMatch(metric, expectedLabels) {
 					gotLabelCount := len(metric.GetLabel())
 					wantLabelCount := len(expectedLabels)
 					if wantLabelCount != gotLabelCount {
 						t.Errorf("Got metric with %d labels, but wanted %d labels. Wanted %#+v for %s",
 							gotLabelCount, wantLabelCount, expectedLabels, metric.String())
 					}
-					return metric
```

Comment:
This line is important. You need to return here otherwise this function will never successfully find a match.

This is why the blaze unit test is failing. You can run these tests locally (`bazel test //staging/src/k8s.io/apiserver/pkg/admission/metrics:go_default_test`) to debug what is happening.

Reply:
right.
thanks